### ISSUE TITLE
addon-a11y: let us choose to set wcag21aaa as our standard

### ISSUE THREAD

**Issue by kdmcguire** [storybookjs/storybook#18768]
**Is your feature request related to a problem? Please describe**
We want to aim for AAA accessibility, but the A11y addon uses AA as the threshold for passes and doesn't show AAA violations.

**Describe the solution you'd like**
There should be a config setting for specifying the standard you'd like A11y to use as your target.

**Describe alternatives you've considered**
We're also looking at how to specify this in jest-axe tests, but having it displayed in the Storybook UI would be a good reassurance for people who browse Storybook but don't run tests (e.g. stakeholders)

**Are you able to assist to bring the feature to reality?**
I wish, but no! I'm not skilled enough.

**Additional context**
We know this component fails AAA colour contrast check, but because it passes AA Storybook is marking it as a pass.
![image](https://user-images.githubusercontent.com/2109807/180404729-c6c289c0-d01c-479a-acab-b07fb8e1c4d4.png)

**Comment by MarcusMorba** [comment]
It should be possible to manage this via custom tag settings:
https://www.deque.com/axe/core-documentation/api-documentation/#axe-core-tags

**Comment by diazemiliano** [comment]
@MarcusMorba at current im doing the following:


``` javascript
// .storybook/preview.js

function enableRulesByTag(tags = []) {
  const allRules = axe.getRules();
  return allRules.map(rule =>
    tags.some(t => rule.tags.includes(t)) ? { id: rule.ruleId, enabled: true } : { id: rule.ruleId, enabled: false }
  );
}

const AxeRules = enableRulesByTag(["wcag2a", "wcag21a", "wcag2aa", "wcag21aa"]);

export const parameters = {
...
  a11y: {
    config: {
      detailedReport: true,
      detailedReportOptions: {
        html: true,
      },
      rules: AxeRules,
    },
  },
...
```

**Comment by jtiala** [comment]
The config in the previous comment isn't correct anymore. What I'm using right now:


```typescript
// preview.ts
import type { A11yParameters } from "@storybook/addon-a11y";
import type { Preview } from "@storybook/react";
import { RuleObject, getRules } from "axe-core";

function getEnabledRulesByTags(tags: string[]): RuleObject {
  const allRules = getRules();

  return allRules.reduce<RuleObject>(
    (ruleObject, rule) => ({
      ...ruleObject,
      [rule.ruleId]: { enabled: tags.some((tag) => rule.tags.includes(tag)) },
    }),
    {},
  );
}

const enabledTags = [
  "wcag2a",
  "wcag2aa",
  "wcag21a",
  "wcag21aa",
  "wcag22aa",
  "best-practice",
];

const a11y: A11yParameters = {
  options: {
    rules: getEnabledRulesByTags(enabledTags),
  },
};

const preview: Preview = {
  parameters: {
    // ...
    a11y,
  },
};

export default preview;
```
